src/python/WMCore/Lexicon.py

```diff
@@ -710,6 +711,103 @@ def activity(candidate):
     raise AssertionError("Invalid dashboard activity: %s should 'test'" % candidate)
 
 
+def gpuParameters(candidate):
+    """
+    Validate the spec "GPUParams" argument, which is a JSON encoded object, thus:
+    * an encoded None object (like 'null')
+    * an encoded dictionary with the following parameters:
+      * mandatory: GPUMemoryMB (int), CUDARuntime (str), CUDACapabilities (list of str)
+      * optional: GPUName (str), CUDADriverVersion (str), CUDARuntimeVersion (str)
+    :param candidate: a JSON encoded data to be validated
+    :return: True if validation succeeded, False or exception otherwise
+    """
+    mandatoryArgs = set(["GPUMemoryMB", "CUDARuntime", "CUDACapabilities"])
+    optionalArgs = set(["GPUName", "CUDADriverVersion", "CUDARuntimeVersion"])
+    try:
+        data = json.loads(candidate)
+    except Exception:
+        raise AssertionError("GPUParams is not a valid JSON object")
```

Comment:
Sorry for the question @amaltaro , but why should we raise Assertion error here? It seems strange to me to get a general exception and raise assertion error instead and from the message it even seems like one related to a specific case related to the data structure itself.

Reply:
Lexicon checks either return False or raise an `AssertionError` in case of failures during the input data validation.
This is also the standard behaviour of the `check` function in this module.